__tests__/habits/habits.test.js

```diff
@@ -205,4 +205,34 @@ describe('API - Habit', () => {
 				});
 		});
 	});
+
+	describe('Habit - Update', () => {
+		test('It should require authorization', () =>
+			request.put('/api/habit/52').then(response => {
+				expect(response.statusCode).toBe(401);
+			}));
+
+		test('If an invalid habit ID is passed a 404 error should be returned', done =>
+			request
+				.put('/api/habit/something')
+				.set('Authorization', token)
+				.end((err, res) => {
+					if (err) throw err;
+					expect(res.status).toBe(404);
+					expect(res.body.message).toBe('Habit not found');
+					done();
+				}));
+
+		test('If a valid habit ID is passed, that habit should be updated', done => {
+			request
+				.put(`/api/habit/${habitId}`)
+				.set('Authorization', token)
+				.end((err, res) => {
+					if (err) throw err;
+					expect(res.status).toBe(200);
+					expect(res.body.success).toBeTruthy();
```

Comment:
Maybe check if res.body[updated value] is actually updated here? Also, I'm not seeing any values that should be updated here, this kind of request shouldn't return success.

Reply:
Thanks. It's done.

Reply:
So this now updates the habit with the same values that it had before. Perhaps you could change some fields so it really is updating the habit.

There are still no tests to check any of the updated values of the habit that gets returned. So checking `res.body.habit.name`, etc.

Reply:
thanks. It skipped my mind.